src/Pages/AddItem.js

```diff
@@ -1,19 +1,28 @@
 import React, { useState } from 'react';
+import { useCollection } from 'react-firebase-hooks/firestore';
 import { db } from '../lib/firebase';
+import ErrorMessage from '../components/ErrorMessage';
 
 export default function AddItem({ token }) {
+  const [list, loading, error] = useCollection(db.collection(token));
+  const [errorMessage, setErrorMessage] = useState('');
   const [item, setItem] = useState('');
   const [urgency, setUrgency] = useState(7);
 
   const handleSubmit = (e) => {
     e.preventDefault();
-    db.collection(token).add({
-      item,
-      urgency,
-      lastPurchased: null,
-    });
-    setUrgency(7);
-    setItem('');
+
+    if (validateNewListItem(item, list)) {
```

Comment:
I'm afraid I led you a little astray here.. generally speaking I like to see validators return 'true' when things are valid, and 'false' when we've found a problem with them. My suggestion of the double bang contradicted this belief. We should be able to resolve this by changing the double bang to just a single exclamation mark, but doing so will require us to flip this if/else around.

Reply:
This to me reads that if our item is valid, we're throwing an error.. even though that's not what the logic is actually doing

Reply:
Gotcha, thanks for clarifying this.